cylc/flow/scripts/cylc.py

```diff
@@ -33,15 +34,30 @@
 )
 
 
+def get_version(long=False):
+    """Return the Cylc Flow version and top level installation path.
+
+    Path reports correctly for normal pip and conda installations
+    but not for "pip install -e ." because of the source links.
```

Comment:
Probably doesn't report correctly for Zip imports (but quite sure it won't work anyway due to namespaces in our packages)

Reply:
How about displaying the path to the `cylc.flow` module rather than the top-level env?

```python
Path(cylc.flow.__file__).parent
```

Reply:
I doubt the rather lengthy cylc-flow Python library location is what users really want to see:
```
$ cylc version --long
8.0a3.dev (/home/oliverh/cylc/cylc-flow/worktrees/cylc-version-long/venv/lib/python3.7/site-packages/cylc/flow)
```
If you're using the CLI (`cylc version`) then you really care more about the application(s), which is much closer to the top:
```
$ which cylc
~/cylc/cylc-flow/worktrees/cylc-version-long/venv/bin/cylc
```
and the common parent location is of course the top level of the venv, which is what I currently have it printing:
```
cylc version --long
8.0a3.dev (/home/oliverh/cylc/cylc-flow/worktrees/cylc-version-long/venv)
```
And bear in mind that the conda version (or even a more sensibly named Python venv) is even clearer:
```
8.0a3.dev (/home/oliverh/miniconda3/envs/cylc-8.0a3.dev)
```
What do you think?

Reply:
(Another way to look at it: print where the *package* is installed, which is more than just the Python library).

Reply:
> Probably doesn't report correctly for Zip imports (but quite sure it won't work anyway due to namespaces in our packages)

I'm sure we can tweak this function in the future, to work for other kinds of installations.

Reply:
> 8.0a3.dev (/home/oliverh/cylc/cylc-flow/worktrees/cylc-version-long/venv/lib/python3.7/site-packages/cylc/flow)

Wow, you've got a funky setup!

> If you're using the CLI (cylc version) then you really care more about the application(s), which is much closer to the top:

In that case how about the path to the Cylc executable:

```
$ which cylc
$ python3 -c 'import cylc.flow.scripts.cylc; print(cylc.flow.scripts.__file__)'
```

Reply:
> In that case how about the path to the Cylc executable:

No, `cylc.flow.scripts.__file__` is the path to the library modules called by the executable. 

> Would using the Python executable location work?

That's a good idea :tada:  It works even for `pip install -e .`.  I'll push up the change...

(Now I hope we don't have to worry about people going to great lengths to run cylc-flow with the system Python :confounded:)